### PR TITLE
[DevASAN][CPU] bugfix for mmap return value check.

### PR DESCRIPTION
**Pull request by yingcong-wu**
`mmap()` would return `MAP_FAILED` instead of NULL in case of error, we need to do a translation here.

Intel/llvm PR: https://github.com/intel/llvm/pull/16466

### CODE REVIEW

**Review by aarongreig — APPROVED** [review]

**Review by zhaomaosu — APPROVED** [review]

File: source/loader/layers/sanitizer/asan/asan_shadow.cpp [hunk 1]
```diff
@@ -16,6 +16,8 @@
 #include "sanitizer_common/sanitizer_utils.hpp"
 #include "ur_sanitizer_layer.hpp"
 
+#include <errno.h>
+
```
Comment (AllanZyne) [inline source/loader/layers/sanitizer/asan/asan_shadow.cpp:20]: "asan_shadow.cpp" is not linux specific code, please remove this.

Comment (yingcong-wu) [reply source/loader/layers/sanitizer/asan/asan_shadow.cpp]: removed.

**Review by AllanZyne — APPROVED** [review]

**Comment by yingcong-wu** [comment]
Hi @aarongreig , could you help merge this fix? Thanks.

**Comment by aarongreig** [comment]
> Hi @aarongreig , could you help merge this fix? Thanks.

Happy to merge right away as it's a trivial change but I'll need you to make the LLVM PR to update the tag. I would do it myself but you can't approve your own PR and there wont be another UR reviewer around to approve until late next week

**Comment by yingcong-wu** [comment]
> > Hi @aarongreig , could you help merge this fix? Thanks.
> 
> Happy to merge right away as it's a trivial change but I'll need you to make the LLVM PR to update the tag. I would do it myself but you can't approve your own PR and there wont be another UR reviewer around to approve until late next week

Thank you, updated.

**Comment by aarongreig** [comment]
@yingcong-wu does this change need the 2025.1 label applied to it?

**Comment by AllanZyne** [comment]
> @yingcong-wu does this change need the 2025.1 label applied to it?

Yep, please add it. Thanks!

**Pull request merged**